quill-core/src/main/scala/io/getquill/context/ActionMacro.scala

```diff
@@ -37,18 +37,25 @@ class ActionMacro(val c: MacroContext) extends ContextMacro with ReifyLiftings {
     translateBatchQueryPrettyPrint(quoted, q"false")
 
   def translateBatchQueryPrettyPrint(quoted: Tree, prettyPrint: Tree): Tree =
-    expandBatchAction(quoted) { case (batch, param, expanded) =>
-      q"""
+    expandBatchActionNew(quoted, false) {
+      case (batch, param, expanded, injectableLiftList, idiomNamingOriginalAstVars, idiomContext, canDoBatch) =>
+        q"""
           ..${EnableReflectiveCalls(c)}
+          val batches =
+            if ($canDoBatch) {
+              $batch.toList.grouped(1).toList
```

Comment:
@deusaquilus can I remove the `if` and keep only this line? 🤔

Reply:
There are certain systems where in some cases we “pretend” to do batch so that’s why this is there. Maybe we removed all those cases but I’m not sure so let’s leave it for now.

Reply:
I don’t think so #2811